custom_components/nordpool/aio_price.py

```diff
@@ -227,10 +227,17 @@ async def fetch(self, data_type, end_date=None, areas=None):
 
         res = await asyncio.gather(*jobs)
 
-        raw = [self._parse_json(i, areas) for i in res]
+        raw = [await self._async_parse_json(i, areas) for i in res]
```

Comment:
It would be more efficient to run the whole list comp in the executor instead of creating a job for each item being iterated

Reply:
True! And goot catch. Honestly I didn't put to much thought in it as it was my first priority to create a quick fix. Actually, a lot that's happening in the integration should be done in a library, but that is a lot more work for which I don't have time atm

Reply:
It wasnt supported in the library. Im the one that created PR for the async interface in nordpool package. At the time it wasnt merged and i didnt want to wait. There were issues like using mutable default arguments.